pages/api/randomProfile.js

```diff
@@ -0,0 +1,40 @@
+// pages/api/randomProfile.js
+require('dotenv').config();
+
+
+const { MongoClient } = require('mongodb');
+
+// console.log('MONGODB_URI:', process.env.MONGODB_URI);
+
+
+const uri = process.env.MONGODB_URI;
+const client = new MongoClient(uri, { useNewUrlParser: true, useUnifiedTopology: true });
+
+export default async function handler(req, res) {
+  if (req.method != 'GET') {
+    return res.status(400).json({ error: 'Invalid request: GET request required' });
+  }
+
+  try {
+    await client.connect();
+    const database = client.db('linkfree');
+
+    // Use the "profiles" collection
+    const collection = database.collection('profiles');
+    const userProfiles = await collection.find({}).toArray();
+    // console.log(userProfiles); // Add this line to log the retrieved profiles
+
+
+    // Select a random profile from the userProfiles array
+    const randomProfile = userProfiles[Math.floor(Math.random() * userProfiles.length)];
```

Comment:
Mongo can do the random selection, rather than selecting fetching all the profiles and doing it in js

Reply:
Implemented!